### DISCUSSION TITLE
How to deploy static website

### DISCUSSION THREAD

**Discussion by vedeja** [likec4/likec4#601] [Q&A] [answered]
I'm building a static website using the CLI `likec4 build`. After that I move the generated folder `dist` to my hosting. LikeC4 runs but says "No diagrams found". I compare locally and see that if I remove the source `.c4` files, the command `likec4 preview -o ./dist` fails. So I add the `.c4` files to my hosting, thinking they are needed for the application actually show diagrams, but the "No diagrams found" remains. I think I'm missing something conceptually here when it comes to deploying my diagrams. Can you help?

**Comment by davydkov** [comment, marked as answer]
Hey @vedeja 

You don't need your source .c4 files after you have built the static website.
Basically, `likec4 build` builds `index.html` and all javascripts to render your model.
You need to copy this index.html and assets folder (everything from `dist/*` to the root of your hosting).

Have you checked this repo?
https://github.com/likec4/template

**Reply by davydkov** [reply]
@vedeja, please note that Vite is used internally by likec4, and `likec4 build` calls `vite build`.

[This page](https://vitejs.dev/guide/static-deploy.html) from vite documentation might also help you.

**Reply by vedeja** [reply]
It seems like building with --use-dot-bin results in lots of:
`[BinaryGraphvizLayouter.layout] command(exit:undefined): 'dot -Tjson -y'
ERROR Error: Command failed with ENOENT: dot -Tjson -y
spawn dot ENOENT` after which the site built has no diagrams. The reason we added --use-dot-bin in the first place was because the build got stuck.

**Reply by davydkov** [reply]
There might be some warnings if you use non-latin characters or icons, but don't worry, they're not errors.
Can you try [v0.60.0](https://github.com/likec4/likec4/releases/tag/v0.60.0)?
This version handles such warnings better.
If after updating your diagrams are still missing, there might be an issue, please let me know.

**Reply by jubr** [reply]
It actually looks like `dot` is not found on the system. Do you have Grahviz installed? Ie for Ubuntu: `apt-get install graphviz`
